src/main/java/com/ibm/eventstreams/connect/mqsource/JMSReader.java

```diff
@@ -93,116 +94,80 @@ public JMSReader() {
+                final String sslTruststoreLocation = config.getString(MQSourceConnector.CONFIG_NAME_MQ_SSL_TRUSTSTORE_LOCATION);
+                final Password sslTruststorePassword = config.getPassword(MQSourceConnector.CONFIG_NAME_MQ_SSL_TRUSTSTORE_PASSWORD);
                 if (sslKeystoreLocation != null || sslTruststoreLocation != null) {
                     final SSLContext sslContext = buildSslContext(sslKeystoreLocation, sslKeystorePassword,
                             sslTruststoreLocation, sslTruststorePassword);
                     mqConnFactory.setSSLSocketFactory(sslContext.getSocketFactory());
                 }
             }
 
-            queue = new MQQueue(queueName);
+            queue = new MQQueue(config.getString(MQSourceConnector.CONFIG_NAME_MQ_QUEUE));
 
-            this.userName = userName;
-            this.password = password;
+            userName = config.getString(MQSourceConnector.CONFIG_NAME_MQ_USER_NAME);
+            password = config.getPassword(MQSourceConnector.CONFIG_NAME_MQ_PASSWORD);
 
-            this.messageBodyJms = false;
-            queue.setMessageBodyStyle(WMQConstants.WMQ_MESSAGE_BODY_MQ);
-            if (mbj != null) {
-                if (Boolean.parseBoolean(mbj)) {
-                    this.messageBodyJms = true;
-                    queue.setMessageBodyStyle(WMQConstants.WMQ_MESSAGE_BODY_JMS);
-                }
-            }
+            messageBodyJms = config.getBoolean(MQSourceConnector.CONFIG_NAME_MQ_MESSAGE_BODY_JMS);
+            queue.setMessageBodyStyle(messageBodyJms ?
+                                        WMQConstants.WMQ_MESSAGE_BODY_JMS :
+                                        WMQConstants.WMQ_MESSAGE_BODY_MQ);
 
-            if (mdr != null) {
-                if (Boolean.parseBoolean(mdr)) {
-                    queue.setBooleanProperty(WMQConstants.WMQ_MQMD_READ_ENABLED, true);
-                }
-            }
+            queue.setBooleanProperty(WMQConstants.WMQ_MQMD_READ_ENABLED,
+                                     config.getBoolean(MQSourceConnector.CONFIG_NAME_MQ_MESSAGE_MQMD_READ));
+
+            topic = config.getString(MQSourceConnector.CONFIG_NAME_TOPIC);
 
-            this.topic = topic;
         } catch (JMSException | JMSRuntimeException jmse) {
             log.error("JMS exception {}", jmse);
             throw new ConnectException(jmse);
+        } catch (final MalformedURLException e) {
+            log.error("MalformedURLException exception {}", e);
+            throw new ConnectException("CCDT file url invalid", e);
         }
 
+
+        final String builderClass = config.getString(MQSourceConnector.CONFIG_NAME_MQ_RECORD_BUILDER);
         try {
             final Class<? extends RecordBuilder> c = Class.forName(builderClass).asSubclass(RecordBuilder.class);
             builder = c.newInstance();
-            builder.configure(props);
+            builder.configure(config.originalsStrings());
```

Comment:
Is there a reason to pass the strings rather than the config itself?

Reply:
I put that in the commit message:

> Note that I did have to leave the existing Map<String, String> parameter in the record builder as that is a public interface that users have subclassed. This should ideally be a Kafka Connect object rather than a raw map, but in the interest of not breaking existing subclasses, I've left it as a map. This does mean I'm converting the properties map to a config object twice, but as this is only at startup I think the performance cost will be minimal.

It is a bit clunky and frustrating - in an ideal world we wouldn't be passing untyped strings around, but without a breaking change I felt like we were stuck with it for now.